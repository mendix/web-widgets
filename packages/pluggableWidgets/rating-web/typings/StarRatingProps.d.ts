/**
 * This file was generated from StarRating.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { CSSProperties } from "react";
import { ActionValue, DynamicValue, EditableValue, WebIcon } from "mendix";
import { Big } from "big.js";

export interface StarRatingContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    rateAttribute: EditableValue<Big>;
    emptyIcon?: DynamicValue<WebIcon>;
    icon?: DynamicValue<WebIcon>;
    maximumStars: number;
    animation: boolean;
    onChange?: ActionValue;
}

export interface StarRatingPreviewProps {
    /**
     * @deprecated Deprecated since version 9.18.0. Please use class property instead.
     */
    className: string;
    class: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    renderMode?: "design" | "xray" | "structure";
    rateAttribute: string;
    emptyIcon: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; iconUrl: string; } | { type: "icon"; iconClass: string; } | undefined;
    icon: { type: "glyph"; iconClass: string; } | { type: "image"; imageUrl: string; iconUrl: string; } | { type: "icon"; iconClass: string; } | undefined;
    maximumStars: number | null;
    animation: boolean;
    onChange: {} | null;
}
