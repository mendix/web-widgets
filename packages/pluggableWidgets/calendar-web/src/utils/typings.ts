import { ObjectItem } from "mendix";
import { CalendarProps as ReactCalendarProps } from "react-big-calendar";
import { withDragAndDropProps } from "react-big-calendar/lib/addons/dragAndDrop";

export interface CalendarEvent {
    title: string;
    start: Date;
    end: Date;
    allDay: boolean;
    color?: string;
    item: ObjectItem;
}

export type EventDropOrResize = {
    event: CalendarEvent;
    start: Date;
    end: Date;
};

export interface DragAndDropCalendarProps<TEvent extends object = Event, TResource extends object = object>
    extends ReactCalendarProps<TEvent, TResource>,
        withDragAndDropProps<TEvent, TResource> {}
