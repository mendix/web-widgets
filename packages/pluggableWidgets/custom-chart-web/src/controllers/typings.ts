import { ActionValue, EditableValue } from "mendix";

export interface ControllerProps {
    configurationOptions: string;
    dataAttribute?: EditableValue<string>;
    dataStatic: string;
    eventDataAttribute?: EditableValue<string>;
    layoutAttribute?: EditableValue<string>;
    layoutStatic: string;
    onClick?: ActionValue;
    sampleData: string;
    sampleLayout: string;
}
