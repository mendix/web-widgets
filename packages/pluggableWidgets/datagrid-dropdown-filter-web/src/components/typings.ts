import { RefFilterStore } from "@mendix/widget-plugin-dropdown-filter/stores/RefFilterStore";
import { StaticSelectFilterStore } from "@mendix/widget-plugin-dropdown-filter/stores/StaticSelectFilterStore";

export interface EnumFilterAPI {
    filterStore: StaticSelectFilterStore;
    parentChannelName?: string;
}

export interface RefFilterAPI {
    filterStore: RefFilterStore;
    parentChannelName?: string;
}
