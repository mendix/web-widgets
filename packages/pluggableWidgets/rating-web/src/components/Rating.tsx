import { CSSProperties, ReactElement, useRef, useState } from "react";
import classNames from "classnames";

export interface RatingProps {
    animated: boolean;
    className?: string;
    disabled: boolean;
    emptyIcon: ReactElement;
    fullIcon: ReactElement;
    maximumValue: number;
    onChange?: (value: number) => void;
    style?: CSSProperties;
    tabIndex?: number;
    value: number;
}

const enum Direction {
    PREVIOUS = "previous",
    NEXT = "next"
}

export function Rating(props: RatingProps): ReactElement {
    const [hover, setHover] = useState<undefined | number>(undefined);
    const containerRef = useRef<HTMLDivElement>(null);
    const onClickAction = !props.disabled
        ? (currentIndex: number): void => {
              if (props.value === currentIndex) {
                  props.onChange?.(0);
              } else {
                  props.onChange?.(currentIndex);
              }
          }
        : undefined;

    const focusItem = (direction: Direction): void => {
        if (containerRef.current) {
            const currentFocusedElement = containerRef.current.querySelector(".rating-item:focus");
            if (direction === Direction.NEXT && currentFocusedElement?.nextSibling) {
                (currentFocusedElement.nextSibling as HTMLDivElement).focus();
            } else if (direction === Direction.PREVIOUS && currentFocusedElement?.previousSibling) {
                (currentFocusedElement.previousSibling as HTMLDivElement).focus();
            }
        }
    };

    return (
        <div
            className={classNames("widget-rating", props.className)}
            data-focusindex={props.tabIndex ?? 0}
            role="radiogroup"
            style={props.style}
            onKeyDown={event => {
                switch (event.key) {
                    case "Left": // Microsoft Edge value
                    case "ArrowLeft":
                        event.preventDefault();
                        focusItem(Direction.PREVIOUS);
                        break;
                    case "Right": // Microsoft Edge value
                    case "ArrowRight":
                        event.preventDefault();
                        focusItem(Direction.NEXT);
                        break;
                }
            }}
            ref={containerRef}
        >
            {Array.from({ length: props.maximumValue }, (_, index) => {
                const currentIndex = index + 1;
                return (
                    <div
                        aria-checked={currentIndex === props.value}
                        aria-label={currentIndex.toString()}
                        className={classNames("rating-item", { disabled: props.disabled })}
                        key={`icon_${currentIndex}`}
                        onClick={() => {
                            setHover(undefined);
                            onClickAction?.(currentIndex);
                        }}
                        onKeyDown={event => {
                            if (event.key === " " || event.key === "Enter") {
                                event.preventDefault();
                                onClickAction?.(currentIndex);
                            }
                        }}
                        onMouseOut={() => {
                            setHover(undefined);
                        }}
                        onMouseOver={() => {
                            if (!props.disabled && props.animated) {
                                setHover(currentIndex);
                            }
                        }}
                        role="radio"
                        tabIndex={index === Math.max(props.value - 1, 0) ? 0 : -1}
                    >
                        {Number(currentIndex) <= props.value ? (
                            props.fullIcon
                        ) : hover && currentIndex <= Number(hover) ? (
                            <div className="rating-item-hover">{props.fullIcon}</div>
                        ) : (
                            props.emptyIcon
                        )}
                    </div>
                );
            })}
        </div>
    );
}
