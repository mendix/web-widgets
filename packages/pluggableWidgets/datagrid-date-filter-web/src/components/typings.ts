import { Date_InputFilterInterface } from "@mendix/widget-plugin-filtering/typings/InputFilterInterface";

export interface DateFilterProps {
    filterStore: Date_InputFilterInterface;
    parentChannelName?: string;
}
