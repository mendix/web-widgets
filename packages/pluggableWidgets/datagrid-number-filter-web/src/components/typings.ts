import { ChangeEventHandler, CSSProperties } from "react";
import { FilterType } from "../../typings/FilterType";
import { Big } from "big.js";

interface FilterProps {
    adjustable: boolean;
    defaultFilterType: FilterType;
    className?: string;
    id?: string;
    placeholder?: string;
    screenReaderButtonCaption?: string;
    screenReaderInputCaption?: string;
    tabIndex?: number;
    styles?: CSSProperties;
}

export interface FilterInputProps extends FilterProps {
    onFilterTypeClick: (type: FilterType) => void;
    onInputChange: ChangeEventHandler<HTMLInputElement>;
    inputValue: string;
    inputRef?: React.ClassAttributes<HTMLInputElement>["ref"];
    inputDisabled?: boolean;
}

export interface FilterComponentProps extends FilterProps {
    inputChangeDelay: number;
    value?: Big;
    updateFilters?: (value: Big | undefined, type: FilterType) => void;
    name: string;
    datagridChannelName?: string;
}
