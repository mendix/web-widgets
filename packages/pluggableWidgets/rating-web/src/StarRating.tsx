import { createElement, ReactElement, useCallback } from "react";
import { Rating as RatingComponent } from "./components/Rating";
import { ValueStatus } from "mendix";
import { executeAction } from "@mendix/widget-plugin-platform/framework/execute-action";
import { isAvailable } from "@mendix/widget-plugin-platform/framework/is-available";
import Big from "big.js";
import { StarRatingContainerProps } from "../typings/StarRatingProps";
import { Icon } from "./components/Icon";

import "./ui/rating-main.scss";

export function StarRating(props: StarRatingContainerProps): ReactElement {
    const emptyIcon =
        props.emptyIcon && isAvailable(props.emptyIcon) ? (
            <Icon value={props.emptyIcon?.value} empty animate={props.animation} />
        ) : (
            <Icon value={{ type: "icon", iconClass: "" }} empty animate={props.animation} />
        );
    const fullIcon =
        props.icon && isAvailable(props.icon) ? (
            <Icon value={props.icon?.value} full animate={props.animation} />
        ) : (
            <Icon value={{ type: "icon", iconClass: "" }} full animate={props.animation} />
        );

    const onChange = useCallback(
        (value: number) => {
            if (props.rateAttribute.status === ValueStatus.Available) {
                props.rateAttribute.setValue(new Big(value));
                executeAction(props.onChange);
            }
        },
        [props.rateAttribute, props.onChange]
    );

    const value = Number(isAvailable(props.rateAttribute) ? props.rateAttribute.value : 0);

    return (
        <RatingComponent
            className={props.class}
            animated={props.animation}
            disabled={props.rateAttribute.readOnly}
            emptyIcon={emptyIcon}
            fullIcon={fullIcon}
            maximumValue={props.maximumStars}
            onChange={onChange}
            tabIndex={props.tabIndex}
            value={value > props.maximumStars ? props.maximumStars : value}
        />
    );
}
