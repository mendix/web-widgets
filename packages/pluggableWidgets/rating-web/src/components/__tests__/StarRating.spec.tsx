import { createElement } from "react";
import { actionValue, EditableValueBuilder } from "@mendix/widget-plugin-test-utils";
import { StarRatingContainerProps } from "../../../typings/StarRatingProps";
import { Big } from "big.js";
import { mount, shallow } from "enzyme";
import { StarRating } from "../../StarRating";
import { Rating } from "../Rating";

describe("Rating Container", () => {
    const defaultProps: StarRatingContainerProps = {
        class: "",
        name: "rating",
        tabIndex: 0,
        rateAttribute: new EditableValueBuilder<Big>().withValue(new Big(5)).build(),
        animation: true,
        maximumStars: 5
    };

    it("renders correctly the structure", () => {
        const rating = shallow(<StarRating {...defaultProps} />);
        expect(rating).toMatchSnapshot();
    });

    it("renders correctly the structure without animation", () => {
        const rating = shallow(<StarRating {...defaultProps} animation={false} />);
        expect(rating).toMatchSnapshot();
    });

    it("renders correctly the structure when disabled", () => {
        const rating = shallow(
            <StarRating
                {...defaultProps}
                rateAttribute={new EditableValueBuilder<Big>().withValue(new Big(1)).isReadOnly().build()}
            />
        );
        expect(rating).toMatchSnapshot();
    });

    describe("with events", () => {
        it("triggers correctly on change action", () => {
            const onChange = actionValue();
            const ratingWrapper = mount(<StarRating {...defaultProps} onChange={onChange} />);
            const rating = ratingWrapper.find(Rating);
            const options = rating.find("div.rating-item");
            options.at(0).simulate("click");

            expect(onChange.execute).toHaveBeenCalled();
        });

        it("defines correct values to attribute on change action", () => {
            const ratingAttribute = new EditableValueBuilder<Big>().withValue(new Big(0)).build();
            const ratingWrapper = mount(<StarRating {...defaultProps} rateAttribute={ratingAttribute} />);
            const rating = ratingWrapper.find(Rating);
            const options = rating.find("div.rating-item");
            options.at(0).simulate("click");

            expect(ratingAttribute.setValue).toHaveBeenCalled();
            expect(ratingAttribute.setValue).toHaveBeenCalledWith(new Big(1));
        });

        it("doesnt call setValue when value is read only", () => {
            const ratingAttribute = new EditableValueBuilder<Big>().withValue(new Big(1)).isReadOnly().build();
            const ratingWrapper = mount(<StarRating {...defaultProps} rateAttribute={ratingAttribute} />);
            const rating = ratingWrapper.find(Rating);
            const options = rating.find("div.rating-item");
            options.at(0).simulate("click");

            expect(ratingAttribute.setValue).toHaveBeenCalledTimes(0);
        });

        it("doesnt trigger on change action when attribute is read only", () => {
            const onChange = actionValue();
            const ratingWrapper = mount(
                <StarRating
                    {...defaultProps}
                    rateAttribute={new EditableValueBuilder<Big>().withValue(new Big(1)).isReadOnly().build()}
                />
            );
            const rating = ratingWrapper.find(Rating);
            const options = rating.find("div.rating-item");
            options.at(0).simulate("click");

            expect(onChange.execute).toHaveBeenCalledTimes(0);
        });
    });
});
