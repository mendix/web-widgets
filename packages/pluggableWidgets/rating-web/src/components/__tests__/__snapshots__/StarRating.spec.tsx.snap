// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarRating rendering renders correctly the structure 1`] = `
<DocumentFragment>
  <div
    class="widget-rating"
    data-focusindex="0"
    role="radiogroup"
  >
    <div
      aria-checked="false"
      aria-label="1"
      class="rating-item"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-full animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="2"
      class="rating-item"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-full animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="3"
      class="rating-item"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-full animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="4"
      class="rating-item"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-full animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="true"
      aria-label="5"
      class="rating-item"
      role="radio"
      tabindex="0"
    >
      <span
        class="rating-icon rating-icon-full animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StarRating rendering renders correctly the structure when disabled 1`] = `
<DocumentFragment>
  <div
    class="widget-rating"
    data-focusindex="0"
    role="radiogroup"
  >
    <div
      aria-checked="true"
      aria-label="1"
      class="rating-item disabled"
      role="radio"
      tabindex="0"
    >
      <span
        class="rating-icon rating-icon-full animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="2"
      class="rating-item disabled"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-empty animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24.59,28.63l-8.54-3.87-8.63,3.86,.97-9.39L2.12,12.34l9.22-2.21L16,1.98l4.65,8.14,9.22,2.21-6.26,6.9,.97,9.4ZM5.88,13.49l4.59,5.06-.7,6.83,6.29-2.81,6.19,2.8-.7-6.82,4.59-5.06-6.78-1.62-3.35-5.86-3.35,5.86-6.78,1.62Z"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="3"
      class="rating-item disabled"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-empty animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24.59,28.63l-8.54-3.87-8.63,3.86,.97-9.39L2.12,12.34l9.22-2.21L16,1.98l4.65,8.14,9.22,2.21-6.26,6.9,.97,9.4ZM5.88,13.49l4.59,5.06-.7,6.83,6.29-2.81,6.19,2.8-.7-6.82,4.59-5.06-6.78-1.62-3.35-5.86-3.35,5.86-6.78,1.62Z"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="4"
      class="rating-item disabled"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-empty animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24.59,28.63l-8.54-3.87-8.63,3.86,.97-9.39L2.12,12.34l9.22-2.21L16,1.98l4.65,8.14,9.22,2.21-6.26,6.9,.97,9.4ZM5.88,13.49l4.59,5.06-.7,6.83,6.29-2.81,6.19,2.8-.7-6.82,4.59-5.06-6.78-1.62-3.35-5.86-3.35,5.86-6.78,1.62Z"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="5"
      class="rating-item disabled"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-empty animate"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24.59,28.63l-8.54-3.87-8.63,3.86,.97-9.39L2.12,12.34l9.22-2.21L16,1.98l4.65,8.14,9.22,2.21-6.26,6.9,.97,9.4ZM5.88,13.49l4.59,5.06-.7,6.83,6.29-2.81,6.19,2.8-.7-6.82,4.59-5.06-6.78-1.62-3.35-5.86-3.35,5.86-6.78,1.62Z"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StarRating rendering renders correctly the structure without animation 1`] = `
<DocumentFragment>
  <div
    class="widget-rating"
    data-focusindex="0"
    role="radiogroup"
  >
    <div
      aria-checked="false"
      aria-label="1"
      class="rating-item"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-full"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="2"
      class="rating-item"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-full"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="3"
      class="rating-item"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-full"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="false"
      aria-label="4"
      class="rating-item"
      role="radio"
      tabindex="-1"
    >
      <span
        class="rating-icon rating-icon-full"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
    <div
      aria-checked="true"
      aria-label="5"
      class="rating-item"
      role="radio"
      tabindex="0"
    >
      <span
        class="rating-icon rating-icon-full"
      >
        <svg
          fill="currentColor"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polygon
            points="29.88 12.34 20.65 10.13 16 1.98 11.35 10.13 2.12 12.34 8.38 19.23 7.42 28.62 16.04 24.76 24.59 28.63 23.62 19.23 29.88 12.34"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
